pkg/endpoint/endpoint.go

```diff
@@ -277,16 +249,56 @@ func (e *EndpointStatus) lastIndex() int {
 	return lastIndex
 }
 
+// getAndIncIdx returns current free slot index and increments the index to the
+// next index that can be overwritten.
 func (e *EndpointStatus) getAndIncIdx() int {
 	idx := e.Index
 	e.Index++
 	if e.Index >= maxLogs {
 		e.Index = 0
 	}
+	// Lets skip the CurrentStatus message from the log to prevent removing
+	// non-OK status!
+	if e.Index < len(e.Log) &&
+		e.Log[e.Index] == e.CurrentStatus &&
+		e.CurrentStatus.Status.Code != OK {
+		e.Index++
+		if e.Index >= maxLogs {
+			e.Index = 0
+		}
+	}
 	return idx
 }
 
-func (e *EndpointStatus) addStatusLog(s *statusLog) {
+// addStatusLog adds statusLogMsg to endpoint log.
```

Comment:
@tgraf Tell me if by only reading this explanation it makes sense to you or if you had any doubts.